core/app/models/spree/order.rb

```diff
@@ -516,7 +516,7 @@ def ensure_shipping_address
     end
 
     def create_proposed_shipments
-      return self.shipments if unreturned_exchange? && shipments.all?(&:shipped?)
+      return self.shipments if unreturned_exchange?
```

Comment:
I think I clarified this in chat, but yes. unreturned_exchanges are a special case which deals with shipments itself.